packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -443,8 +443,9 @@ export class CreateQueryEditorStore extends QueryEditorStore {
       ),
     );
     if (this.classPath) {
-      this.queryBuilderState.querySetupState._class =
-        this.queryBuilderState.graphManagerState.graph.getClass(this.classPath);
+      this.queryBuilderState.changeClass(
+        this.queryBuilderState.graphManagerState.graph.getClass(this.classPath),
+      );
       this.queryBuilderState.querySetupState.setClassIsReadOnly(true);
     } else {
       this.queryBuilderState.querySetupState._class = getNullableFirstElement(
```

Comment:
what happens if the class selected here is milestoned. Should we also use the `changeClass` method here